src/service/chain_rewardable_entities.proto

```diff
@@ -0,0 +1,203 @@
+// TODO: Design for different key formats - track "archetype" of key ... helium
+// vs. solana, network prefixes etc.
+// TODO: Consider OTHER Rewardable Entities (REs) in the future
+// TODO: Signature STUFF in req's and reports
+
+syntax = "proto3";
+
+package helium.chain_rewardable_entities;
+
+message solana_pub_key {
+  // As string - base58 encode
+  bytes value = 1;
+}
+
+message helium_pub_key {
+  // As string - base58 check encode helium pubkey
+  bytes value = 1;
+}
+
+// ------------------------------------------------------------------
+// Helium Hotspot "rewardable entities" and their metadata
+// ------------------------------------------------------------------
+enum mobile_hotspot_device_type {
+  mobile_hotspot_device_type_UNKNOWN =
+      0; // Unknown device type, should not be used
+  mobile_hotspot_device_type_CBRS = 1;
+  mobile_hotspot_device_type_WIFI_INDOOR = 2;
+  mobile_hotspot_device_type_WIFI_OUTDOOR = 3;
+  mobile_hotspot_device_type_WIFI_DATA_ONLY = 4;
+}
+
+message mobile_hotspot_metadata {
+  // Serial number of the mobile hotspot device (can be null or empty... in the
+  // case of data-only devices)
+  string serial_number = 2;
+  // Type of the mobile hotspot device
+  mobile_hotspot_device_type device_type = 3;
+  // The asserted location in hex format
+  string asserted_hex = 4;
+  uint32 azimuth = 5;
+}
+
+// TODO: this meets mvp app needs, but could easily be extended
+message iot_hotspot_metadata {
+  // The asserted hex location of the IoT hotspot
+  string asserted_hex = 2;
+  uint32 elevation = 3;
+  bool is_data_only = 4;
+}
+
+// ------------------------------------------------------------------
+// Rewardable Entity <---> Controlling Wallet
+// ------------------------------------------------------------------
+enum entity_owner_type {
+  // Unknown control type, should not be used
+  entity_owner_type_UNKNOWN = 0;
+  // Owner of the NFT
+  entity_owner_type_DIRECT_OWNER = 1;
+  // Controlling permission of the Welcome Pack
+  entity_owner_type_WELCOME_PACK_OWNER = 2;
+}
+
+message entity_owner_info {
+  // User wallet public key
+  solana_pub_key wallet = 1;
+
+  // Type of control relationship
+  entity_owner_type type = 2;
+}
+
+message rewards_split_v1 {
+  // They key of the on-chain struct that contains/performs the split
+  solana_pub_key pub_key = 1;
+  // The cron schedule of the split
+  string schedule = 2;
+  // The total shares of the split (user can specify any number of shares,
+  // defaults to 100)
+  uint32 total_shares = 3;
```

Comment:
This is a question for my own curiosity. 

How do we ensure `split_recipient_info_v1` shares don't exceed `total_shares`?

What happens if `split_recipient_info_v1` have fixed amount of rewards, but not enough rewards are issued in a given epoch?